llm/src/flash_attn_triton.py

```diff
@@ -0,0 +1,1112 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Triton implementation of Flash Attention.
```

Comment:
Just want to double check on this file, is it a direct copy of https://github.com/HazyResearch/flash-attention/blob/main/flash_attn/flash_attn_triton.py?

And if so, is it up to date (last change was 2 weeks ago)?

Reply:
this is from the [bert dir](https://github.com/mosaicml/examples/blob/main/bert/src/flash_attn_triton.py)

I guess it points to the need for a unified code dir.

Reply:
@abhi-mosaic Yes, but we have not been incorporating changes to that original implementation into the BERT implementation.
Agreed that a location for unified code would be advantageous. I support the choice to re-use the BERT code for now, since it will be easier to refactor things if it's just a matter of reducing redundancy.

Reply:
perhaps we can link to the specific commit it is up to date with, so we don't lose track of that.

Reply:
It would also be good to call out with comments anywhere that we have made changes. Not sure what the longer term plan is for keeping this in sync (or not) with hazy research.

Reply:
I symlinked to the BERT version of this file (but git doesn't respect symlinks)
changes can be see in this PRs changes to `bert/src/flash_attn_triton.py`